app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1207,23 +1261,28 @@ public void onLoadingFailed(final String imageUri, final View view,
     @Override
     public void onLoadingComplete(final String imageUri, final View view,
                                   final Bitmap loadedImage) {
-        final float width = Math.min(
+        // scale the notification thumbnail
```

Comment:
scale to what and why?

Reply:
Performance improvements. IIRC if we pass the full size thumbnail to the notification, the notification needs more RAM and every time it is updated, the thumbnail needs to the scaled again. That's why we scale it down to the maximum thumbnail size.

Reply:
Then expand the comment to something like `scale down the notification thumbnail for performance`